src/celeritas/track/detail/ProcessSecondariesLauncher.hh

```diff
@@ -128,6 +128,7 @@ ProcessSecondariesLauncher<M>::operator()(TrackSlotId tid) const
             ti.geo.dir = secondary.direction;
             ti.particle.particle_id = secondary.particle_id;
             ti.particle.energy = secondary.energy;
+            CELER_ASSERT(ti);
```

Comment:
Since the `Secondary`'s operator bool just uses the particle ID, I think we should assert on line 106 that `secondary.energy > zero_quantity()` and probably `is_soft_unit_vector(secondary.direction)`.